lib/storage/data/file/DataFileStore.js

```diff
@@ -256,18 +256,15 @@ class DataFileStore {
                     { method: 'put', key, filePath, error: err });
                 // destroying the write stream forces a close(fd)
                 fileStream.destroy();
+                // we need to unlink the file ourselves
+                fs.stat(filePath, err => {
+                    if (!err) {
+                        fs.unlinkSync(filePath);
+                    }
+                });
                 return cbOnce(errors.InternalError.customizeDescription(
                     `read stream error: ${err.code}`));
             });
-            dataStream.on('close', () => {
```

Comment:
Close is now always called and not just called when an issue comes up, https://github.com/nodejs/node/pull/20611

Reply:
@vrancurel Would you mind having a look at this one? Thank you.
@miniscruff Could you please create a JIRA ticket to add tests for this? Thank you.

Reply:
ARSN-52 Done